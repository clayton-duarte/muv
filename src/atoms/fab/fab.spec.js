import React from 'react';

import Fab from '.';

it('should render a fab', () => {
  const wrapper = mount(
    <Fab />,
  );
  expect(wrapper).toMatchSnapshot();
});
