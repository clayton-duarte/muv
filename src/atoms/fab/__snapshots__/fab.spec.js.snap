// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a fab 1`] = `
<Fab
  onClick={[Function]}
>
  <Styled(Button)
    onClick={[Function]}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".5rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "gsbsDV",
            "rules": Array [
              "
box-shadow: ",
              [Function],
              ";
bottom: ",
              [Function],
              ";
right: ",
              [Function],
              ";
justify-content: center;
align-items: center;
border-radius: 50%;
position: fixed;
height: 3.5rem;
display: flex;
width: 3.5rem;
padding: 0;
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "color": Object {
                "black": "#222222",
                "darkgray": "#515151",
                "error": "#dd3333",
                "gray": "#a9abae",
                "lightgray": "#e5e5e5",
                "overlay": "rgba(0, 0, 0, .15)",
                "primary": "#0081c8",
                "secondary": "#00245f",
                "success": "#33bb33",
                "warn": "#dd3333",
                "white": "#fefefe",
              },
              "font": Object {
                "color": "#222222",
                "family": "\\"Ubuntu\\", sans-serif",
                "size": "16px",
                "spacing": 0,
              },
              "shape": Object {
                "border": ".125rem solid rgba(0, 0, 0, .15)",
                "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                "margin": ".25rem .125rem",
                "padding": ".75rem .875rem",
                "radius": ".5rem",
                "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                "size": "16px",
              },
              "transition": Object {
                "time": ".3s ease",
              },
            },
          },
          "displayName": "Styled(Button)",
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      theme={
        Object {
          "color": Object {
            "black": "#222222",
            "darkgray": "#515151",
            "error": "#dd3333",
            "gray": "#a9abae",
            "lightgray": "#e5e5e5",
            "overlay": "rgba(0, 0, 0, .15)",
            "primary": "#0081c8",
            "secondary": "#00245f",
            "success": "#33bb33",
            "warn": "#dd3333",
            "white": "#fefefe",
          },
          "font": Object {
            "color": "#222222",
            "family": "\\"Ubuntu\\", sans-serif",
            "size": "16px",
            "spacing": 0,
          },
          "shape": Object {
            "border": ".125rem solid rgba(0, 0, 0, .15)",
            "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
            "margin": ".25rem .125rem",
            "padding": ".75rem .875rem",
            "radius": ".5rem",
            "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
            "size": "16px",
          },
          "transition": Object {
            "time": ".3s ease",
          },
        }
      }
    >
      <Button
        className="sc-iwsKbI gsbsDV"
        disabled={false}
        href={null}
        inset={false}
        onClick={[Function]}
        outline={false}
        secondary={false}
        theme={
          Object {
            "color": Object {
              "black": "#222222",
              "darkgray": "#515151",
              "error": "#dd3333",
              "gray": "#a9abae",
              "lightgray": "#e5e5e5",
              "overlay": "rgba(0, 0, 0, .15)",
              "primary": "#0081c8",
              "secondary": "#00245f",
              "success": "#33bb33",
              "warn": "#dd3333",
              "white": "#fefefe",
            },
            "font": Object {
              "color": "#222222",
              "family": "\\"Ubuntu\\", sans-serif",
              "size": "16px",
              "spacing": 0,
            },
            "shape": Object {
              "border": ".125rem solid rgba(0, 0, 0, .15)",
              "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
              "margin": ".25rem .125rem",
              "padding": ".75rem .875rem",
              "radius": ".5rem",
              "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
              "size": "16px",
            },
            "transition": Object {
              "time": ".3s ease",
            },
          }
        }
      >
        <styled.button
          className="sc-iwsKbI gsbsDV"
          disabled={false}
          href={null}
          inset={false}
          onClick={[Function]}
          outline={false}
          secondary={false}
          theme={
            Object {
              "color": Object {
                "black": "#222222",
                "darkgray": "#515151",
                "error": "#dd3333",
                "gray": "#a9abae",
                "lightgray": "#e5e5e5",
                "overlay": "rgba(0, 0, 0, .15)",
                "primary": "#0081c8",
                "secondary": "#00245f",
                "success": "#33bb33",
                "warn": "#dd3333",
                "white": "#fefefe",
              },
              "font": Object {
                "color": "#222222",
                "family": "\\"Ubuntu\\", sans-serif",
                "size": "16px",
                "spacing": 0,
              },
              "shape": Object {
                "border": ".125rem solid rgba(0, 0, 0, .15)",
                "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                "margin": ".25rem .125rem",
                "padding": ".75rem .875rem",
                "radius": ".5rem",
                "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                "size": "16px",
              },
              "transition": Object {
                "time": ".3s ease",
              },
            }
          }
        >
          <StyledComponent
            className="sc-iwsKbI gsbsDV"
            disabled={false}
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": false,
                  "lastClassName": "hTeMnD",
                  "rules": Array [
                    "
",
                    [Function],
                    "
",
                  ],
                },
                "defaultProps": Object {
                  "theme": Object {
                    "color": Object {
                      "black": "#222222",
                      "darkgray": "#515151",
                      "error": "#dd3333",
                      "gray": "#a9abae",
                      "lightgray": "#e5e5e5",
                      "overlay": "rgba(0, 0, 0, .15)",
                      "primary": "#0081c8",
                      "secondary": "#00245f",
                      "success": "#33bb33",
                      "warn": "#dd3333",
                      "white": "#fefefe",
                    },
                    "font": Object {
                      "color": "#222222",
                      "family": "\\"Ubuntu\\", sans-serif",
                      "size": "16px",
                      "spacing": 0,
                    },
                    "shape": Object {
                      "border": ".125rem solid rgba(0, 0, 0, .15)",
                      "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                      "margin": ".25rem .125rem",
                      "padding": ".75rem .875rem",
                      "radius": ".5rem",
                      "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                      "size": "16px",
                    },
                    "transition": Object {
                      "time": ".3s ease",
                    },
                  },
                },
                "displayName": "styled.button",
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href={null}
            inset={false}
            onClick={[Function]}
            outline={false}
            secondary={false}
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "\\"Ubuntu\\", sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".5rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
          >
            <button
              className="sc-iwsKbI gsbsDV sc-htoDjs hTeMnD"
              disabled={false}
              href={null}
              onClick={[Function]}
            >
              <Text
                align="center"
                isLabel={false}
                small={false}
                strong={false}
                transform="none"
                type=""
                white={true}
              >
                <styled.p
                  align="center"
                  isLabel={false}
                  small={false}
                  strong={false}
                  theme={
                    Object {
                      "color": Object {
                        "black": "#222222",
                        "darkgray": "#515151",
                        "error": "#dd3333",
                        "gray": "#a9abae",
                        "lightgray": "#e5e5e5",
                        "overlay": "rgba(0, 0, 0, .15)",
                        "primary": "#0081c8",
                        "secondary": "#00245f",
                        "success": "#33bb33",
                        "warn": "#dd3333",
                        "white": "#fefefe",
                      },
                      "font": Object {
                        "color": "#222222",
                        "family": "\\"Ubuntu\\", sans-serif",
                        "size": "16px",
                        "spacing": 0,
                      },
                      "shape": Object {
                        "border": ".125rem solid rgba(0, 0, 0, .15)",
                        "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                        "margin": ".25rem .125rem",
                        "padding": ".75rem .875rem",
                        "radius": ".5rem",
                        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                        "size": "16px",
                      },
                      "transition": Object {
                        "time": ".3s ease",
                      },
                    }
                  }
                  transform="none"
                  white={true}
                >
                  <StyledComponent
                    align="center"
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": false,
                          "lastClassName": "kOSZjX",
                          "rules": Array [
                            "
font-size: ",
                            [Function],
                            ";
",
                            [Function],
                            "
",
                          ],
                        },
                        "defaultProps": Object {
                          "theme": Object {
                            "color": Object {
                              "black": "#222222",
                              "darkgray": "#515151",
                              "error": "#dd3333",
                              "gray": "#a9abae",
                              "lightgray": "#e5e5e5",
                              "overlay": "rgba(0, 0, 0, .15)",
                              "primary": "#0081c8",
                              "secondary": "#00245f",
                              "success": "#33bb33",
                              "warn": "#dd3333",
                              "white": "#fefefe",
                            },
                            "font": Object {
                              "color": "#222222",
                              "family": "\\"Ubuntu\\", sans-serif",
                              "size": "16px",
                              "spacing": 0,
                            },
                            "shape": Object {
                              "border": ".125rem solid rgba(0, 0, 0, .15)",
                              "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                              "margin": ".25rem .125rem",
                              "padding": ".75rem .875rem",
                              "radius": ".5rem",
                              "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                              "size": "16px",
                            },
                            "transition": Object {
                              "time": ".3s ease",
                            },
                          },
                        },
                        "displayName": "styled.p",
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    isLabel={false}
                    small={false}
                    strong={false}
                    theme={
                      Object {
                        "color": Object {
                          "black": "#222222",
                          "darkgray": "#515151",
                          "error": "#dd3333",
                          "gray": "#a9abae",
                          "lightgray": "#e5e5e5",
                          "overlay": "rgba(0, 0, 0, .15)",
                          "primary": "#0081c8",
                          "secondary": "#00245f",
                          "success": "#33bb33",
                          "warn": "#dd3333",
                          "white": "#fefefe",
                        },
                        "font": Object {
                          "color": "#222222",
                          "family": "\\"Ubuntu\\", sans-serif",
                          "size": "16px",
                          "spacing": 0,
                        },
                        "shape": Object {
                          "border": ".125rem solid rgba(0, 0, 0, .15)",
                          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
                          "margin": ".25rem .125rem",
                          "padding": ".75rem .875rem",
                          "radius": ".5rem",
                          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                          "size": "16px",
                        },
                        "transition": Object {
                          "time": ".3s ease",
                        },
                      }
                    }
                    transform="none"
                    white={true}
                  >
                    <p
                      className="sc-bdVaJa kOSZjX"
                      transform="none"
                    >
                      FAB
                    </p>
                  </StyledComponent>
                </styled.p>
              </Text>
            </button>
          </StyledComponent>
        </styled.button>
      </Button>
    </StyledComponent>
  </Styled(Button)>
</Fab>
`;
